### ISSUE TITLE
Publish to nuget?

### ISSUE THREAD

**Issue by dougajmcdonald** [alphagov/notifications-net-client#38]
Could this be published as a nuget package? would make it a lot easier to consume
  
Apologies I did miss the BinTray reference, this is currently blocked for us so will have to get it whitelisted before we can use it.

**Comment by fabiomilheiro** [comment]
Yes, I realise that this is already 9 months old but a nuget package is what we should have here. I am not using that bintray thing after looking about specific environment variables (!). Don't want unnecessary set ups for a web API client. For now, I am using my own gov notify client.

**Comment by quis** [comment]
This seems like a good idea if Nuget is what most .NET teams use. However it’ll take us some time to get round to doing the work to publish there. In the mean time does whitelisting Bintray work for you?

**Comment by fabiomilheiro** [comment]
@quis, thanks for your response. I don't know about others, but our project will not be using bintray at all. There are too many package sources already and we need to focus on value to the client.

Our environments are already working with a test service with an API client that we built based on your .NET client code available in github. Thanks for sharing that code.

When you have available a nuget package, we'd be much appreciated if you update this thread so we can start using it though.

Cheers!

**Comment by manos-msdn** [comment]
The issue with bintray solution (besides what already mentioned about network config and whitelisting) is that,  virtually 100% of .NET projects include nuget as a source, and the way nuget works, adding a 2nd source (bintray) just for the notify package, means nuget will, at random, request ALL packages from EITHER source (to be exact will chose the one with lowest ping, but that can be random when all sources are on equally good networks). There is no way to prioritise/designate which source should be used for each package. Therefore, assuming there is a solution with 100+ packages, all except "notify" hosted on nuget, means several of those packages might end up being requested from bintray, resulting in "not found" which delays the builds. This looks horrible on the build log.

There is virtually no good reason to keep bintray. Please: 

A) choose a more specific name than "Notify" e.g. like UK.Gov.Notify or somehting
B) publish on nuget

Lots of people will be happy with this

**Comment by idavidmcdonald** [comment]
Hi all,

Just thought I'd check in on this open issue. We haven't been able to dedicate any time to looking into moving from Bintray to Nuget I'm afraid. Would love to hear if this is still an issue for you or from anyone new to add their voice (even if it's just a thumbs up to confirm this is something you want to see happen) so we can try and prioritise higher (although I can't promise anything for sure).

Thanks for your patience :)

**Comment by manos-msdn** [comment]
This will be an issue until you resolve it, obviously fairly low priority for most people as all it does is add unnecessarily time to the build. 

The solution to this should be trivial really, just open an account on nuget (or use an already existing one) and change the target place where you upload your nugets. However give that this is open now for 2.5 years, I feel like a safe bet is, it will be faster for someone to clone this and publish a nuget that you guys do it, sorry :/

**Comment by idavidmcdonald** [comment]
Hi, an update on this from us.

Sorry for the delay but you will be pleased to hear that our hand has been forced to prioritise this as Bintray is being deprecated on the 1st of May.

We are in the process of moving to nuget.org. We’ve currently only got version 4.0.0 (the most recent version) hosted there but hope to get key older versions backdated too.

We've had to change the package name from `Notify` to `GovukNotify` as someone had already taken `Notify` on nuget.org but have tried to make this as small a change as possible for you.

We will be putting proper comms out for all users by email with full instructions likely some time next week.

You can find the package at https://www.nuget.org/packages/GovukNotify. We welcome any early thoughts as we will be continuing on with this over the next couple of days.

**Comment by idavidmcdonald** [comment]
Closing this as the package is now available on Nuget, we've sent comms out to all our .net users asking them to move to use Nuget and have had no problems reported.

Apologies for the (major) delay again but hopefully this is now better for everyone :)
